core-processor/src/main/java/io/micronaut/inject/processing/DeclaredBeanElementCreator.java

```diff
@@ -358,27 +365,36 @@ private boolean visitInjectAndLifecycleMethod(BeanDefinitionVisitor visitor, Met
         if (claimed) {
             return true;
         }
-        if (!methodElement.isStatic() && isInjectPointMethod(methodElement)) {
+        if (!methodElement.isStatic() && isInjectPointMethod(propertyElement, methodElement)) {
             staticMethodCheck(methodElement);
             // TODO: Require @ReflectiveAccess for private methods in Micronaut 4
-            visitMethodInjectionPoint(visitor, methodElement);
+            if (propertyElement != null) {
+                // use the properties metadata for property elements
+                FieldElement fieldElement = propertyElement.getField().orElse(null);
+                if (fieldElement != null) {
```

Comment:
Why do we need to work with field here? It should be handled by the field visiting